### PR TITLE
cleanup CI dependencies

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
## Summary of changes

Changes introduced in this pull request:

- get rid of multiple occurrences in our CI where we needlessly queried Canonical repositories - GH images are pretty self-contained and have most of the tools needed. https://github.com/actions/runner-images?tab=readme-ov-file#software-and-image-support
- removed unused dependencies

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by LesnyRumcajs** [comment]
I removed some dependencies from MacOS build as well; it might fail when out of draft.

File: .github/workflows/forest.yml [hunk 1]
```diff
@@ -68,13 +68,7 @@ jobs:
         uses: mozilla-actions/sccache-action@v0.0.4
         timeout-minutes: '${{ fromJSON(env.CACHE_TIMEOUT_MINUTES) }}'
         continue-on-error: true
-      - name: Install Apt Dependencies
-        run: |
-          sudo make install-deps
       - name: Cargo Install
-        env:
-          # To minimize compile times: https://nnethercote.github.io/perf-book/build-configuration.html#minimizing-compile-times
-          RUSTFLAGS: "-C linker=clang -C link-arg=-fuse-ld=lld"
```
Comment (hanabi1224) [inline .github/workflows/forest.yml:77]: I think clang is preinstalled, and this RUSTFLAGS saves ~1min linking time.

https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#language-and-runtime

File: .github/workflows/forest.yml [hunk 2]
```diff
@@ -21,8 +21,8 @@ env:
   AWS_ACCESS_KEY_ID: '${{ secrets.AWS_ACCESS_KEY_ID }}'
   AWS_SECRET_ACCESS_KEY: '${{ secrets.AWS_SECRET_ACCESS_KEY }}'
   RUSTC_WRAPPER: sccache
-  CC: sccache clang
```
Comment (hanabi1224) [inline .github/workflows/forest.yml:24]: Do we have to switch to gcc? clang should be working without the apt step as it's preinstalled in the runner image
https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#language-and-runtime

File: .github/workflows/forest.yml (hunk 1 above)
Comment (LesnyRumcajs) [reply .github/workflows/forest.yml:77]: According to docs it should, though the build failed on missing dependency because of this. Let me reproduce it.

Comment (hanabi1224) [reply .github/workflows/forest.yml]: Ah I see, I guess it failed because the llvm toolchain is missing (required by `cc`)

**Review by hanabi1224 — APPROVED** [review]

File: .github/workflows/forest.yml (hunk 1 above)
Comment (LesnyRumcajs) [reply .github/workflows/forest.yml]: Hm, actually there is no error. I'll re-run it a few times just in case.

**Review by ruseinov — APPROVED** [review]
looks good, `api compare` is failing, but that's probably just flakiness. I can't see how these changes would affect that job, especially given the fact that it's containerised.

**Comment by LesnyRumcajs** [comment]
> looks good, `api compare` is failing, but that's probably just flakiness. I can't see how these changes would affect that job, especially given the fact that it's containerised.

The failures are unrelated and addressed in another PR.

**Pull request merged**